### PR TITLE
e2e_apimachinery: stop using deprecated framework.ExpectEqual

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

e2e_apimachinery: stop using deprecated framework.ExpectEqual

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jiahuif** [comment]
/lgtm
Thank you.
/assign @jpbetz
for approval

**Review by jiahuif — APPROVED** [review]

File: test/e2e/apimachinery/aggregator.go [hunk 1]
```diff
@@ -516,7 +518,7 @@ func TestSampleAPIServer(ctx context.Context, f *framework.Framework, aggrclient
 	var jr *apiregistrationv1.APIService
 	err = json.Unmarshal([]byte(statusContent), &jr)
 	framework.ExpectNoError(err, "Failed to process statusContent: %v | err: %v ", string(statusContent), err)
-	framework.ExpectEqual(jr.Status.Conditions[0].Message, "all checks passed", "The Message returned was %v", jr.Status.Conditions[0].Message)
+	gomega.Expect(jr.Status.Conditions[0].Message).To(gomega.Equal("all checks passed"), "The Message returned was %v", jr.Status.Conditions[0].Message)
```
Comment (pohly) [inline test/e2e/apimachinery/aggregator.go:521]: There are ways to make statements about a collection (`jr.Status.Conditions` in this case`), but let's not get carried away - this is fine.

File: test/e2e/apimachinery/aggregator.go [hunk 2]
```diff
@@ -616,7 +618,7 @@ func TestSampleAPIServer(ctx context.Context, f *framework.Framework, aggrclient
 			framework.Logf("Observed APIService %v with Labels: %v & Condition: %v", wardle.ObjectMeta.Name, wardle.Labels, cond)
 		}
 	}
-	framework.ExpectEqual(foundUpdatedStatusCondition, true, "The updated status condition was not found. %#v", wardle.Status.Conditions)
+	gomega.Expect(foundUpdatedStatusCondition).To(gomega.BeTrue(), "The updated status condition was not found. %#v", wardle.Status.Conditions)
```
Comment (pohly) [inline test/e2e/apimachinery/aggregator.go:621]: `if !foundUpdatedStatusCondition framework.Failf("The updated status condition was not found in:\n%s", format.Object(wardle.Status.Conditions, 1))`

will produce a more readable failure message.

File: test/e2e/apimachinery/aggregator.go [hunk 3]
```diff
@@ -714,7 +716,7 @@ func TestSampleAPIServer(ctx context.Context, f *framework.Framework, aggrclient
 			framework.Logf("Observed APIService %v with Labels: %v & Conditions: %v", wardle.ObjectMeta.Name, wardle.Labels, cond)
 		}
 	}
-	framework.ExpectEqual(foundPatchedStatusCondition, true, "The patched status condition was not found. %#v", wardle.Status.Conditions)
+	gomega.Expect(foundPatchedStatusCondition).To(gomega.BeTrue(), "The patched status condition was not found. %#v", wardle.Status.Conditions)
```
Comment (pohly) [inline test/e2e/apimachinery/aggregator.go:719]: Same here.

File: test/e2e/apimachinery/crd_conversion_webhook.go [hunk 4]
```diff
@@ -461,17 +461,17 @@ func testCRListConversion(ctx context.Context, f *framework.Framework, testCrd *
 	list, err := customResourceClients["v1"].List(ctx, metav1.ListOptions{})
 	framework.ExpectNoError(err)
 	gomega.Expect(len(list.Items)).To(gomega.BeIdenticalTo(2))
-	framework.ExpectEqual((list.Items[0].GetName() == name1 && list.Items[1].GetName() == name2) ||
-		(list.Items[0].GetName() == name2 && list.Items[1].GetName() == name1), true)
+	gomega.Expect((list.Items[0].GetName() == name1 && list.Items[1].GetName() == name2) ||
+		(list.Items[0].GetName() == name2 && list.Items[1].GetName() == name1)).To(gomega.BeTrue())
```
Comment (pohly) [inline test/e2e/apimachinery/crd_conversion_webhook.go:465]: This is basically checking that a list contains two elements in arbitrary order, and identifies the elements by their name. What we want to see if this fails is the `list.Items`.

Does this work?
```
Expect(list.Items).To(ConsistOf(HaveField("GetName()", Equal(name1)), HaveField("GetName()", Equal(name2)))
```

Checking the length separately is not necessary when using `ConsistOf`.

File: test/e2e/apimachinery/crd_conversion_webhook.go [hunk 5]
```diff
@@ -461,17 +461,17 @@ func testCRListConversion(ctx context.Context, f *framework.Framework, testCrd *
...
 	verifyV1Object(crd, &list.Items[1])
 
 	ginkgo.By("List CRs in v2")
 	list, err = customResourceClients["v2"].List(ctx, metav1.ListOptions{})
 	framework.ExpectNoError(err)
 	gomega.Expect(len(list.Items)).To(gomega.BeIdenticalTo(2))
-	framework.ExpectEqual((list.Items[0].GetName() == name1 && list.Items[1].GetName() == name2) ||
-		(list.Items[0].GetName() == name2 && list.Items[1].GetName() == name1), true)
+	gomega.Expect((list.Items[0].GetName() == name1 && list.Items[1].GetName() == name2) ||
+		(list.Items[0].GetName() == name2 && list.Items[1].GetName() == name1)).To(gomega.BeTrue())
```
Comment (pohly) [inline test/e2e/apimachinery/crd_conversion_webhook.go:474]: Same here.

File: test/e2e/apimachinery/table_conversion.go [hunk 6]
```diff
@@ -165,7 +165,7 @@ var _ = SIGDescribe("Servers with support for Table transformation", func() {
 		}
 		err := c.AuthorizationV1().RESTClient().Post().Resource("selfsubjectaccessreviews").SetHeader("Accept", "application/json;as=Table;v=v1;g=meta.k8s.io").Body(sar).Do(ctx).Into(table)
 		gomega.Expect(err).To(gomega.HaveOccurred(), "failed to return error when posting self subject access review: %+v, to a backend that does not implement metadata", sar)
-		framework.ExpectEqual(err.(apierrors.APIStatus).Status().Code, int32(406))
+		gomega.Expect(err.(apierrors.APIStatus).Status().Code).To(gomega.Equal(int32(406)))
```
Comment (pohly) [inline test/e2e/apimachinery/table_conversion.go:168]: If this fails, showing the entire error may be useful. How about this:
```
Expect(err.(apierrors.APIStatus)).To(HaveFieled("Status().Code", Equal(int32(406)))
```

File: test/e2e/apimachinery/crd_conversion_webhook.go (hunk 4 above)
Comment (carlory) [reply test/e2e/apimachinery/crd_conversion_webhook.go]: @pohly It does not work as expected. please see

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/119459/pull-kubernetes-e2e-kind-ipv6/1698967903245701120

Comment (pohly) [reply test/e2e/apimachinery/crd_conversion_webhook.go]: Well, it was worth a try :sweat_smile:  Please revert back to the more complicated separate calls.

Looks like an issue with Gomega (cannot call GetName?). I'll try to reproduce and file an issue.

Comment (carlory) [reply test/e2e/apimachinery/crd_conversion_webhook.go]: @pohly I pushed some changes, please review again. 

https://github.com/kubernetes/kubernetes/compare/01905fe68da8efcc714122aed38507a2787a414d..15a402eefb7258f05da6a8bcb807ebe129f326d1

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Comment by aojea** [comment]
@pohly this broke one test https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20detect-local-interface-name-prefix

```


Kubernetes e2e suite: [It] [sig-api-machinery] Namespaces [Serial] should apply changes to a namespace status [Conformance] expand_less | 0s
-- | --
{ failed [FAILED] updated object should have the approved condition, got []v1.NamespaceCondition{v1.NamespaceCondition{Type:"StatusPatch", Status:"True", LastTransitionTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Reason:"E2E", Message:"Patched by an e2e test"}, v1.NamespaceCondition{Type:"StatusUpdate", Status:"True", LastTransitionTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Reason:"E2E", Message:"Updated by an e2e test"}} Expected     <v1.NamespaceConditionType>: StatusUpdate to equal     <string>: StatusUpdate In [It] at: test/e2e/apimachinery/namespace.go:360 @ 09/06/23 16:10:03.833 }


```

**Comment by pohly** [comment]
There's a type difference (`<v1.NamespaceConditionType>` vs `<string>`).

@carlory: can you follow up with a fix?

**Comment by carlory** [comment]
@pohly I will do it.

File: test/e2e/apimachinery/namespace.go [hunk 7]
```diff
@@ -357,9 +356,9 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
 			return err
 		})
 		framework.ExpectNoError(err, "failed to update namespace status %s", ns)
-		framework.ExpectEqual(len(statusUpdated.Status.Conditions), len(statusUpdated.Status.Conditions), fmt.Sprintf("updated object should have the applied condition, got %#v", statusUpdated.Status.Conditions))
-		framework.ExpectEqual(string(statusUpdated.Status.Conditions[len(statusUpdated.Status.Conditions)-1].Type), "StatusUpdate", fmt.Sprintf("updated object should have the approved condition, got %#v", statusUpdated.Status.Conditions))
```
Comment (pohly) [inline test/e2e/apimachinery/namespace.go:361]: The typecast to string was dropped here.

Comment (carlory) [reply test/e2e/apimachinery/namespace.go:361]: how about it?

```
gomega.Expect(statusUpdated.Status.Conditions[len(statusUpdated.Status.Conditions)-1].Type).To(gomega.Equal(v1.NamespaceConditionType("StatusUpdate")), "updated object should have the approved condition, got %#v", statusUpdated.Status.Conditions)
```

Comment (carlory) [reply test/e2e/apimachinery/namespace.go:361]: I raise a pr to fix it. https://github.com/kubernetes/kubernetes/pull/120479

**Pull request merged**